pkg/codegen/go/gen_program.go

```diff
@@ -202,9 +267,32 @@ func (g *generator) genTemps(w io.Writer, temps []interface{}) {
 			args := stripInputs(t.Value.Args[0])
 			g.Fgenf(w, "%.v)\n", args)
 			g.Fgenf(w, "if err != nil {\n")
-			g.Fgenf(w, "return err\n")
+			if genZeroValueDecl {
+				g.Fgenf(w, "return _zero, err\n")
```

Comment:
Another way to do this is `*new(T)`, which is maybe a bit more obscure, but has the benefit of avoiding the variable declaration.

Reply:
Ah, a good tip for the tool belt. I think it might be favorable to avoid obscure syntax for code snippets and docs. Will leave as is unless you have other objections.